runtime/src/main/java/org/corfudb/runtime/LogReplicationServiceClient.java

```diff
@@ -0,0 +1,339 @@
+package org.corfudb.runtime;
+
+import com.google.common.base.Strings;
+import com.google.protobuf.Message;
+import com.google.protobuf.Timestamp;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.LogReplication.ClientDestinationInfoKey;
+import org.corfudb.runtime.LogReplication.ClientRegistrationId;
+import org.corfudb.runtime.LogReplication.ClientRegistrationInfo;
+import org.corfudb.runtime.LogReplication.DestinationInfoVal;
+import org.corfudb.runtime.collections.CorfuStore;
+import org.corfudb.runtime.collections.Table;
+import org.corfudb.runtime.collections.TableOptions;
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError;
+import org.corfudb.util.retry.ExponentialBackoffRetry;
+import org.corfudb.util.retry.IRetry;
+
+import java.lang.reflect.InvocationTargetException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+import static org.corfudb.runtime.view.TableRegistry.CORFU_SYSTEM_NAMESPACE;
+
+/**
+ * A client for log replication
+ */
+@Slf4j
+public class LogReplicationServiceClient {
+
+    public static final String LR_REGISTRY_TABLE_NAME = "LogReplicationRegistrationTable";
```

Comment:
A small request: it is possible to add the name of the protobuf message file to each of these (like `ClientDestinationInfoKey` and `ClientDestinationInfoValue` that will help with readability since we can easily see what this table is used to store early on?
And vice versa - add the table name to the protobuf file?